service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -190,104 +125,51 @@ internal class SøknadsbehandlingServiceIverksettTest {
-            on { publiserUtbetaling(any()) } doReturn utbetalingsRequest.right()
-        }
-        val behandlingMetricsMock = mock<BehandlingMetrics>()
-        val vedtakRepoMock = mock<VedtakRepo>()
-        val kontrollsamtaleServiceMock = mock<KontrollsamtaleService> {
-            on { opprettPlanlagtKontrollsamtale(any(), any()) } doReturn kontrollsamtale().right()
-        }
-        val statistikkObserver = mock<EventObserver>()
-
-        val attesteringstidspunkt = fixedTidspunkt
-
         val serviceAndMocks = SøknadsbehandlingServiceAndMocks(
-            søknadsbehandlingRepo = søknadsbehandlingRepoMock,
-            utbetalingService = utbetalingServiceMock,
-            behandlingMetrics = behandlingMetricsMock,
-            vedtakRepo = vedtakRepoMock,
-            kontrollsamtaleService = kontrollsamtaleServiceMock,
-            observer = statistikkObserver,
+            søknadsbehandlingRepo = mock {
+                on { hent(any()) } doReturn behandling
+            },
+            utbetalingService = mock {
+                on { genererUtbetalingsRequest(any(), any(), any(), any(), any()) } doReturn simulertUtbetaling.right()
+                on { publiserUtbetaling(any()) } doReturn utbetalingsRequest.right()
+            },
+            kontrollsamtaleService = mock {
+                on { opprettPlanlagtKontrollsamtale(any(), any()) } doReturn kontrollsamtale().right()
+            },
         )
+
         val response = serviceAndMocks.søknadsbehandlingService.iverksett(
             SøknadsbehandlingService.IverksettRequest(
                 behandlingId = behandling.id,
-                attestering = Attestering.Iverksatt(attestant, attesteringstidspunkt),
+                attestering = Attestering.Iverksatt(attestant, fixedTidspunkt),
             ),
         )
 
-        val expected = Søknadsbehandling.Iverksatt.Innvilget(
-            id = behandling.id,
-            opprettet = behandling.opprettet,
-            sakId = behandling.sakId,
-            saksnummer = behandling.saksnummer,
-            søknad = behandling.søknad,
-            oppgaveId = behandling.oppgaveId,
-            behandlingsinformasjon = behandling.behandlingsinformasjon,
-            fnr = behandling.fnr,
-            beregning = behandling.beregning,
-            simulering = behandling.simulering,
-            saksbehandler = behandling.saksbehandler,
-            attesteringer = Attesteringshistorikk.empty()
-                .leggTilNyAttestering(Attestering.Iverksatt(attestant, attesteringstidspunkt)),
-            fritekstTilBrev = "",
-            stønadsperiode = behandling.stønadsperiode,
-            grunnlagsdata = Grunnlagsdata.IkkeVurdert,
-            vilkårsvurderinger = Vilkårsvurderinger.Søknadsbehandling.IkkeVurdert,
-        )
+        val expected = søknadsbehandlingIverksattInnvilget().second.copy(id = behandling.id)
```

Comment:
Hmm :thinking: Dette kan jo potensielt være litt skjørt. Men jeg ser kodelinjebesparelsen :]

Reply:
Gitt at testdataene henger sammen så skal det være greit. Så må jo ikke testdataene endres da selvsagt😛  Grunnen til den copy-id tingen er at id-blir generert hver gang, og vil derfor bli diff

Reply:
Trenger vi egentlig å matche hele objektet, eller kunne det holde at det er av riktig type f.eks.?

Reply:
Det er jo servicen som blant annet er ansvarlig for å sette riktig attestant. Men dersom det byttes til å bruke en copy, vil jo ikke denne testen avdekke det. Så jeg er for at vi sjekker på hele objektet, spesielt når det er så enkelt. Da får vi og testet sammenhengen mellom testdataene 🤷

Reply:
Vi kan ha egne tester på at spesifikke ting blir satt på objektet, sånn som f.eks. riktig attestant. Det er kanskje en detalj, men jeg vil nok heller at vi har spesifikke tester enn at det ser ut som vi matcher på hele objektet litt tilfeldig.

Reply:
Jeg likte denne jeg :) Den løser i hvert fall at a) en test feiler dersom uønskede felter endres og b) en test feiler dersom vi ikke legger til attestanten som kom fra routen. Her burde vi nok heller brukt parametre inn mot funksjonen (id+attestant).

Reply:
Ja, tenker det er en filleting mot et par andre tilfeller vi har i testene, men helt generelt ville jeg heller at vi hadde hatt det sånn at spesifikk test X, Y og Z feiler dersom hhv. X, Y og Z går galt, enn at generell test A feiler hvis enten X, Y eller Z går galt. 

Men tenker det er OK sånn, jeg hadde ikke tatt opp denne tingen hvis vi ikke hadde startet diskusjonen rundt "new opp hele objektet" vs. copy.

Reply:
Dersom vi deler opp testene på denne måten er vi da priset at testene X,Y og Z har samme setup (typisk samme private funksjon). Noe som også gir mening i forbindelse med refaktoreringer.